drivers/sht3x/sht3x.c

```diff
@@ -0,0 +1,442 @@
+    }
+
+    /* in single shot mode upate dmeasurement started flag of the driver */
+    if (dev->mode == SHT3X_SINGLE_SHOT) {
+        dev->meas_started = false;
+    }
+    /* start next measurement cycle in periodic modes */
+    else {
+        dev->meas_start_time = xtimer_now_usec();
+        dev->meas_duration = SHT3X_MEASURE_PERIOD[dev->mode];
+    }
+
+    /* check temperature crc */
+    if (_crc8(raw_data,2) != raw_data[2]) {
+        DEBUG_DEV("CRC check for temperature data failed", dev);
+        return -SHT3X_ERROR_CRC;
+    }
+
+    /* check humidity crc */
+    if (_crc8(raw_data+3,2) != raw_data[5]) {
+        DEBUG_DEV("CRC check for humidity data failed", dev);
+        return -SHT3X_ERROR_CRC;
+    }
+
+    return SHT3X_OK;
+}
+
+
+static int _compute_values (uint8_t* raw_data, int16_t* temp, int16_t* hum)
+{
+    if (temp) {
+        uint32_t _tmp = ((uint32_t)raw_data[0] << 8) + raw_data[1];  /* S_T */
+        _tmp  *= 17500; /* S_T x 175 scaled to hundredths of degree */
+        _tmp >>= 16;    /* S_T x 175 / 65535 (shift inaccuracy < resolution) */
+        _tmp  -= 4500;  /* S_T x 175 / 65535 - 45 */
+        *temp = _tmp;
+    }
+
+    if (hum) {
+        uint32_t _tmp = ((uint32_t)raw_data[3] << 8) + raw_data[4]; /* S_RH  */
+        _tmp  *= 10000; /* S_RH x 100 scaled to hundredths of degree */
+        _tmp >>= 16;    /* S_RH x 100 / 65535 (shift inaccuracy < resolution) */
+        *hum = _tmp;
+    }
+
+    return SHT3X_OK;
+}
+
+
+static int _send_command(sht3x_dev_t* dev, uint16_t cmd)
+{
+    ASSERT_PARAM (dev != NULL);
+
+    int res = SHT3X_OK;
+
+    uint8_t data[2] = { cmd >> 8, cmd & 0xff };
+    DEBUG_DEV("send command 0x%02x%02x", dev, data[0], data[1]);
+
+    if (i2c_acquire(dev->i2c_dev) != 0) {
+        DEBUG_DEV ("could not aquire I2C bus", dev);
+        return -SHT3X_ERROR_I2C;
+    }
+
+    res = i2c_write_bytes(dev->i2c_dev, dev->i2c_addr, (const void*)data, 2, 0);
+    i2c_release(dev->i2c_dev);
+
+    if (res != 0) {
+        DEBUG_DEV("could not send command 0x%02x%02x to sensor, reason=%d",
+                  dev, data[0], data[1], res);
+        return -SHT3X_ERROR_I2C;
+    }
+
+    return SHT3X_OK;
+}
+
+
+static int _read_data(sht3x_dev_t* dev, uint8_t *data, uint8_t len)
+{
+    int res = SHT3X_OK;
+
+    if (i2c_acquire(dev->i2c_dev) != 0) {
+        DEBUG_DEV ("could not aquire I2C bus", dev);
+        return -SHT3X_ERROR_I2C;
+    }
+
+    res = i2c_read_bytes(dev->i2c_dev, dev->i2c_addr, (void*)data, len, 0);
+    i2c_release(dev->i2c_dev);
+
+    if (res == 0) {
+#if ENABLE_DEBUG
+        printf("[sht3x] %s bus=%d addr=%02x: read following bytes: ",
+               __func__, dev->i2c_dev, dev->i2c_addr);
+        for (int i=0; i < len; i++)
+            printf("%02x ", data[i]);
+        printf("\n");
+#endif /* ENABLE_DEBUG */
+    }
+    else {
+        DEBUG_DEV("could not read %d bytes from sensor, reason %d (%s)",
+                  dev, len, res, strerror(-res));
```

Comment:
`#include <string.h>` is missing in order to use `strerror()`

Reply:
Strange. It is there in line 21.

Reply:
Ok, it seems that avr-gcc doesn't provide strerror in string.h :worried:

I will remove the strerror output, especially since it seems to map the wrong messages to the error code on some platforms. For example, on my platform an arbitration lost error EAGAIN gives "No more processes". Therefore, it is probably better not to try to print an error message.

Reply:
Done